bundle-workflow/src/manifests/manifest.py

```diff
@@ -19,6 +19,28 @@ def from_path(cls, path):
         with open(path, "r") as f:
             return cls.from_file(f)
 
+    @classmethod
+    def compact(cls, d):
+        clean = {}
+        for k, v in d.items():
+            if isinstance(v, dict):
+                nested = cls.compact(v)
+                if len(nested.keys()) > 0:
+                    clean[k] = nested
+            elif v is not None and v != []:
```

Comment:
Can be check like `if not nested[k]`

Reply:
Not sure I follow, LMK if https://github.com/opensearch-project/opensearch-build/pull/465#:~:text=Resolve%20conversation-,Addressed%20comments.,-%E2%80%A6 can be improved further.

Reply:
Since we are checking whether the key has any value in the dictionary. Why not to check it like `if not dict[key]` which will check for value mapped to the key? Let me know what you think

Reply:
I don't think I understand ;) We already have a hold on `v`, the value through the iteration above. If that value is `None` or `[]` we don't want to write it. Could you please write the function as you suggest?

Reply:
Ohh. I missed this https://github.com/opensearch-project/opensearch-build/commit/1664903d1850bce165e180efc023affcae4523db commit. That exactly does what I suggested above.